compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1419,7 +1419,73 @@ OMR::ResolvedMethodSymbol::getNumberOfBackEdges()
    return numBackEdges;
    }
 
+/*
+ * Returns the bytecode info for an OSR point, taking into account the
+ * possibility that the treetop for a call has different BCI to the OSR
+ * point.
+ */
+TR_ByteCodeInfo
+OMR::ResolvedMethodSymbol::getOSRByteCodeInfo(TR::Node *node)
+   {
+   TR_ByteCodeInfo nodeBCI = node->getByteCodeInfo();
+   if (node->getNumChildren() > 0 && (node->getOpCodeValue() == TR::treetop || node->getOpCode().isCheck()))
+      nodeBCI = node->getFirstChild()->getByteCodeInfo();
+   return nodeBCI;
+   }
+
+/*
+ * Checks the provided node is pending push store or load expected to be
+ * around an OSR point.
+ *
+ * Pending push stores are used to ensure the stack can be reconstructed 
+ * after the transition, whilst anchored loads are also related as they
+ * may be used to avoid the side effects of the prior mentioned stores.
+ */
+bool
+OMR::ResolvedMethodSymbol::isOSRRelatedNode(TR::Node *node)
+   {
+   return (node->getOpCode().isStoreDirect()
+         && node->getOpCode().hasSymbolReference()
+         && node->getSymbolReference()->getSymbol()->isPendingPush())
+      || (node->getOpCodeValue() == TR::treetop
+         && node->getFirstChild()->getOpCode().isLoadVarDirect()
+         && node->getFirstChild()->getOpCode().hasSymbolReference()
+         && node->getFirstChild()->getSymbolReference()->getSymbol()->isPendingPush());
+   }
+
+/*
+ * Checks that the provided node is related to the OSR point's BCI.
+ * This applies to pending push stores and loads sharing the BCI.
+ */
+bool
+OMR::ResolvedMethodSymbol::isOSRRelatedNode(TR::Node *node, TR_ByteCodeInfo osrBCI)
```

Comment:
since the osrBCI is not modified should this be a reference type as well to save a copy?